File: Logger_Service/Models/LogEvent.cs
public class LogEvent
{
    public string ServiceName { get; set; }
    public string Level { get; set; }
    public string Action { get; set; }
    public string Message { get; set; }

    public long? UserId { get; set; }
    public string Role { get; set; }
    public string Email { get; set; }

    public string Method { get; set; }
    public string Path { get; set; }

    public object RequestBody { get; set; }
    public object ResponseBody { get; set; }

    public string Exception { get; set; }
    public DateTime Timestamp { get; set; }
}

File: booking_service/src/main/java/com/plannora/booking_service/logger/DTO/LogEvent.java
package com.plannora.booking_service.logger.DTO;

import lombok.Builder;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@Builder
public class LogEvent {

    private String serviceName;   // user-service
    private String level;         // INFO / ERROR
    private String action;        // LOGIN, REGISTER, UPDATE_PROFILE
    private String message;

    private Long userId;
    private String role;
    private String email;

    private String method;
    private String path;

    private Object requestBody;
    private Object responseBody;

    private String exception;

    private LocalDateTime timestamp;
}
